app/src/main/java/pl/ludek/smat/home/pool_office_client/data/apiservice/ApiModule.kt

```diff
@@ -0,0 +1,10 @@
+package pl.ludek.smat.home.pool_office_client.data.apiservice
+
+object ApiModule  {
+   private const val BASE_URL = "http://10.0.2.2:8080/"
+
+    val retrofitService: ApiPoolInfoService
+        get() = PoolInfoClientNetworkModule.getClient(BASE_URL).create(ApiPoolInfoService::class.java)
```

Comment:
This approach is incorrect. The `ApiModule` is a singleton, which is good, on the other hand you don't store the object created by `.create(ApiPoolInfoService::class.java)` method. In this case the expensive `.create(ApiPoolInfoService::class.java)` method is executed everytime you call `ApiModule.retrofitService`. Much better would be to make `retrofitService` a lazy val, like this:
```
val retrofitService: ApiPoolInfoService by lazy {
    PoolInfoClientNetworkModule.getClient(BASE_URL).create(ApiPoolInfoService::class.java)
}
```
In this case the expensive `.create(ApiPoolInfoService::class.java)` method will be executed only once during the entire app lifetime.
As I said above however, the content of this class will be moved to `PoolInfoClient` class and modified a bit, so please treat my comment just as an explanation/advice.

Reply:
Well, I just thought of fixing it differently. I thought to create one copy of this class in mine activity, and already pass one copy to the methods. Then I would not have created it many times.

Reply:
class OffRelayAll: SingleUseCaseButton {
    private var client = PoolInfoClient

I think it's just not true that I create my own new variable in each case.

Reply:
Although, in my opinion, Java will not create a new class each time - it will probably only be assigned a reference to a pointer to a place in memory. But this is not accurate, I already remember thioria very badly.

Reply:
@Sergey90K in the case you described, the new object would be created when calling `client.retrofitService` not in the place you pointed out.